greenpithumb/db_retriever.py

```diff
@@ -0,0 +1,40 @@
+class DbRetrieverBase(object):
+    """Base class for retrieving information from a database."""
+
+    def __init__(self, cursor):
+        """Creates a new database retrieval object.
+
+        Args:
+            cursor: An SQLite database cursor.
+        """
+        self._cursor = cursor
+
+
+class TemperatureRetriever(DbRetrieverBase):
+    """Retrieves timestamp and temperature readings."""
+
+    def retrieve_temperature(self, range_start, range_end):
+        """Returns timestamp and temperature info within a given range.
+
+        Args:
+            range_start: A datetime.datetime object representing the beginning
+                of the retrieval range.
+            range_end: A datetime.datetime object representing the end of the
+                retrieval range.
+
+        Returns:
+            A list of dictionaries, one per table row, of the timestamp and
+            temperature information.
+        """
+        self._cursor.execute('''SELECT * FROM temperature WHERE timestamp >= ?
+                             and timestamp <= ?''' (range_start, range_end))
+        temperature_data = self._cursor.fetchall()
+
+        temperature_data_list = []
+        for row in temperature_data:
+            d = {}
+            d['timestamp'] = row[0]
```

Comment:
I think we should parse these back into Python objects. The frontend will eventually convert it to JSON and the code here is convenient because it's very close to JSON, but I think we should keep it as an object for as long as possible so that this library stays generic and the frontend is what handles JSON stuff.

So, like:

```python
class TemperatureRecord(object):

  def __init__(self, temperature, timestamp):
    self._temperature = temperature
    self._timestamp = timestamp

  @property
  def temperature(self):
    return self._temperature

  @property
  def timestamp(self):
    return self._timestamp
```

But that's going to get super tedious for all the different record types, so I think we can use named tuples:

```python
TemperatureRecord = collections.namedtuple('TemperatureRecord', ['temperature', 'timestamp'])
```

https://docs.python.org/2/library/collections.html#namedtuple-factory-function-for-tuples-with-named-fields
(the documentation also specifically talks about deserializing from sqlite)

Also, we should probably update the code in the `store_` functions to explicitly convert dates to strings. I suspect the sqlite3 library probably just does the conversion for us silently, but we should do it explicitly because otherwise our deserialization logic depends on sqlite's serialization logic that we don't really see (and theoretically could change).